src/components/Collapsible/Collapsible.tsx

```diff
@@ -30,165 +24,68 @@ export interface CollapsibleProps {
-      this.setState({
-        animationState: 'idle',
-      });
-
+  useEffect(() => {
+    if (open === isOpen) {
       return;
     }
 
-    requestAnimationFrame(() => {
-      const heightNode = this.heightNode.current;
-      switch (animationState) {
-        case 'idle':
-          break;
-        case 'measuring':
-          this.setState({
-            animationState: wasOpen ? 'closingStart' : 'openingStart',
-            height: wasOpen && heightNode ? heightNode.scrollHeight : 0,
-          });
-          break;
-        case 'closingStart':
-          this.setState({
-            animationState: 'closing',
-            height: 0,
-          });
-          break;
-        case 'openingStart':
-          this.setState({
-            animationState: 'opening',
-            height: heightNode ? heightNode.scrollHeight : 0,
-          });
-      }
-    });
-  }
+    setHeight(collapisbleContainer.current.scrollHeight);
+  }, [open]);
 
-  render() {
-    const {id, expandOnPrint, open, children, transition} = this.props;
-    const {animationState, height} = this.state;
-    const parentCollapsibleExpanding = this.context;
-
-    const animating = animationState !== 'idle';
-
-    const wrapperClassName = classNames(
-      styles.Collapsible,
-      open && styles.open,
-      animating && styles.animating,
-      !animating && open && styles.fullyOpen,
-      expandOnPrint && styles.expandOnPrint,
-    );
-
-    const displayHeight = collapsibleHeight(open, animationState, height);
-
-    const content = animating || open || expandOnPrint ? children : null;
-
-    const transitionProperties = transition
-      ? {
-          transitionDuration: `${transition.duration}`,
-          transitionTimingFunction: `${transition.timingFunction}`,
-        }
-      : null;
-
-    return (
-      <ParentCollapsibleExpandingContext.Provider
-        value={
-          parentCollapsibleExpanding || (open && animationState !== 'idle')
-        }
-      >
-        <div
-          id={id}
-          aria-hidden={!open}
-          style={{
-            maxHeight: `${displayHeight}`,
-            ...transitionProperties,
-          }}
-          className={wrapperClassName}
-          ref={this.node}
-          onTransitionEnd={this.handleTransitionEnd}
-        >
-          <div ref={this.heightNode}>{content}</div>
-        </div>
-      </ParentCollapsibleExpandingContext.Provider>
-    );
-  }
-
-  private handleTransitionEnd = (event: TransitionEvent) => {
-    const {target} = event;
-    if (target === this.node.current) {
-      this.setState({animationState: 'idle', height: null});
+  useEffect(() => {
+    if (open || height === null) {
+      return;
     }
-  };
-}
 
-function collapsibleHeight(
-  open: boolean,
-  animationState: AnimationState,
-  height?: number | null,
-) {
-  if (animationState === 'idle' && open) {
-    return open ? 'none' : undefined;
-  }
+    getComputedStyle(collapisbleContainer.current).height;
```

Comment:
This is needed otherwise the rendering engine measures and applies the height then set the new height causing no transition. This causes the rendering engine to apply the 0 height on the next cycle causing the transition to run.